pyproject.toml

```diff
@@ -1,3 +1,36 @@
+[tool.poetry]
+name = "sceptre-resolver-template"
+version = "0.0.1"
+packages = [{ include = "resolver" }]
+readme = "README.md"
+homepage = "https://github.com/Sceptre/sceptre-resolver-template"
+repository = "https://github.com/Sceptre/sceptre-resolver-template"
+authors = ["Sceptre <sceptreorg@gmail.com>"]
+description = "A Github template for Sceptre resolvers"
+keywords = ["sceptre", "sceptre-resolver"]
+license = "Apache-2.0"
+classifiers = [
+  "Intended Audience :: Developers",
+  "Natural Language :: English",
+  "Environment :: Console",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10"
+]
+
+[tool.poetry.plugins."sceptre.resolvers"]
+"custom" = "resolver.custom:Custom"
+
+[tool.poetry.dependencies]
+python = "^3.8.1"
```

Comment:
Don't we need to support Python 3.7? Have we officially ended support for 3.7 yet?

Reply:
meh.. python 3.7 EOL is this June.

Reply:
From the perspective of someone regularly trapped in older versions of Python, I'd prefer we don't de-support any version of Python-3 until there's a new feature we're not prepared to live without.

Reply:
this is a resolver guys, is it really that big of a deal?   sceptre core is not dropping 3.7 just yet.

Reply:
With that said, I think probably we need to do something more specific on the python version. Something like `">=3.7.0,3.11"`

Reply:
Got the below error when attempting to set python version to `>=3.7.0,<3.11`.  It looks like pre-commit has already moved on.   I rather just go with `>=3.8` then change it to that after this june.  Setting to `>=3.8,<3.11` was it suggested works.
 
```
The current project's Python requirement (>=3.7.0,<3.11) is not compatible with some of the required packages Python requirement:
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8
  - pre-commit requires Python >=3.8, so it will not be satisfied for Python >=3.7.0,<3.8

Because no versions of pre-commit match >3.0.0,<3.0.1 || >3.0.1,<3.0.2 || >3.0.2,<3.0.3 || >3.0.3,<3.0.4 || >3.0.4,<3.1.0 || >3.1.0,<3.1.1 || >3.1.1,<3.2.0 || >3.2.0,<3.2.1 || >3.2.1,<3.2.2 || >3.2.2,<4.0.0
 and pre-commit (3.0.0) requires Python >=3.8, pre-commit is forbidden.
And because pre-commit (3.0.1) requires Python >=3.8
 and pre-commit (3.0.2) requires Python >=3.8, pre-commit is forbidden.
And because pre-commit (3.0.3) requires Python >=3.8
 and pre-commit (3.0.4) requires Python >=3.8, pre-commit is forbidden.
And because pre-commit (3.1.0) requires Python >=3.8
 and pre-commit (3.1.1) requires Python >=3.8, pre-commit is forbidden.
And because pre-commit (3.2.0) requires Python >=3.8
 and pre-commit (3.2.1) requires Python >=3.8, pre-commit is forbidden.
So, because pre-commit (3.2.2) requires Python >=3.8
 and sceptre-resolver-template depends on pre-commit (^3.0.0), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties

    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
    For pre-commit, a possible solution would be to set the `python` property to ">=3.8,<3.11"
```